Sources/KituraNet/ClientRequest.swift

```diff
@@ -535,11 +535,12 @@ public class ClientRequest {
      */
     public func end(close: Bool = false) {
         closeConnection = close
-
+        let percentEncodedURL = URL(string: self.percentEncodedURL)!
+        
         var isHTTPS = false
 
         let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-        if (URL(string: percentEncodedURL)?.scheme)! == "https" {
+        if (percentEncodedURL.scheme)! == "https" {
```

Comment:
Same here, even if it was already forced before.

Reply:
Yeah, I don't know how you'd recover from a missing scheme?  But I'll take another look... maybe there's a better way.

Reply:
If we'd have a huge concrete wall of tests, I'd be happy with it, but in its current state my lack of understanding and the code I see scares me too. 

But I don't have any better approaches right now either.

Personally I think force unwrapping is fine if there are multiple good quality and self validating unit tests which reason about the implementation and is convincing of its safety.

Otherwise I'd take the route of being super-defensive; and log warnings & errors whenever the sad path kicks in.

Reply:
My goal right now is just to get tests consistently passing.  So if you can agree that I at least haven't introduced any new problems, can we save the larger restructuring until later?

Reply:
Curiously the existing code does not use URL() as it's base way of storing URLs.  I think one reason for some of the complexity in the code is the way Kitura handles http/https vs Unix domain sockets.  If we're keeping this functionality (which I think is nice to have), it might make some of the code easier to follow one of these (non-standard) conventions for naming Unix domain sockets:  https://github.com/whatwg/url/issues/577